nexus-ingest/nexus-sink/src/main/java/org/nasa/jpl/nexus/ingest/nexussink/DynamoStore.java

```diff
@@ -0,0 +1,56 @@
+package org.nasa.jpl.nexus.ingest.nexussink;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.document.DynamoDB;
+import com.amazonaws.services.dynamodbv2.document.Item;
+import com.amazonaws.services.dynamodbv2.document.Table;
+import org.nasa.jpl.nexus.ingest.wiretypes.NexusContent;
+
+import java.util.Collection;
+
+/**
+ * Created by djsilvan on 6/26/17.
+ */
+public class DynamoStore implements DataStore {
+
+    private DynamoDB dynamoDB;
+    private String tableName;
+    private String primaryKey = "tile_id";
+    private int count = 0;
+
+    public DynamoStore(AmazonDynamoDB dynamoClient, String tableName) {
+        dynamoDB = new DynamoDB(dynamoClient);
+        this.tableName = tableName;
+    }
+
+    public void saveData(Collection<NexusContent.NexusTile> nexusTiles) {
+
+        Table table = dynamoDB.getTable(tableName);
+
+        for (NexusContent.NexusTile tile : nexusTiles) {
+            String tileId = getTileId(tile);
+            byte[] tileData = getTileData(tile);
+
+            try {
+                table.putItem(new Item().withPrimaryKey(primaryKey, tileId).withBinary("data", tileData));
+                count++;
+            }
+            catch (Exception e) {
```

Comment:
Need to improve error handling here. At a minimum these should be turned into logging statements, not `System.err` calls.

I think the appropriate thing to do in this case is raise a custom exception (e.g. `DataStoreException`) and make sure to use the thrown exception as the cause of the custom exception. 

I think that would be good enough for now, but if you want to get fancy:

The NexusTileService should catch `DataStoreException`s and when it catches one try to remove the metadata object (Solr Document). Then it should re-throw a custom `NexusTileServiceException` which could be caught by the NexusSink object and logged.